tests/Security/SessionDownloadStrategyTest.php

```diff
@@ -90,4 +91,81 @@ public function testInvalidArgumentException(): void
 
         new SessionDownloadStrategy($translator, 'foo', 1);
     }
+
+    /**
+     * @group legacy
+     * NEXT_MAJOR: remove this method
+     */
+    public function testLegacyIsGrantedFalseWithGreaterSessionTime(): void
+    {
+        $translator = $this->createMock(LegacyTranslatorInterface::class);
+        $media = $this->createMock(MediaInterface::class);
+        $request = $this->createMock(Request::class);
+        $session = $this->createMock(Session::class);
+
+        $session
+            ->method('get')
+            ->willReturn(1);
+
+        $strategy = new SessionDownloadStrategy($translator, $session, 0);
+        $this->assertFalse($strategy->isGranted($media, $request));
+    }
+
+    /**
+     * @group legacy
+     * NEXT_MAJOR: remove this method
+     */
+    public function testLegacyIsGrantedTrue(): void
+    {
+        $translator = $this->createMock(LegacyTranslatorInterface::class);
+        $media = $this->createMock(MediaInterface::class);
+        $request = $this->createMock(Request::class);
+        $session = $this->createMock(Session::class);
+
+        $session
+            ->method('get')
+            ->willReturn(0);
+
+        $strategy = new SessionDownloadStrategy($translator, $session, 1);
+        $this->assertTrue($strategy->isGranted($media, $request));
+    }
+
+    /**
+     * @group legacy
+     * NEXT_MAJOR: remove this method
+     */
+    public function testLegacyIsGrantedTrueWithContainer(): void
+    {
+        $container = $this->createMock(ContainerInterface::class);
+
+        $translator = $this->createMock(LegacyTranslatorInterface::class);
+        $media = $this->createMock(MediaInterface::class);
+        $request = $this->createMock(Request::class);
+        $session = $this->createMock(Session::class);
+
+        $session
+            ->method('get')
+            ->willReturn(0);
+
+        $container->expects($this->once())
+            ->method('get')
+            ->willReturn($session);
+
+        $strategy = new SessionDownloadStrategy($translator, $container, 1);
+
+        $this->assertTrue($strategy->isGranted($media, $request));
+    }
+
+    /**
+     * @group legacy
+     * NEXT_MAJOR: remove this method
+     */
+    public function testLegacyInvalidArgumentException(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+
+        $translator = $this->createMock(LegacyTranslatorInterface::class);
```

Comment:
```suggestion
        $translator = $this->createStub(LegacyTranslatorInterface::class);
```

Reply:
I copy existing tests. Should we update both?

Reply:
Also `createStub` for non configured object and `createMock` for object where some methods is call?

Reply:
I think that could be a nice improvement :+1: